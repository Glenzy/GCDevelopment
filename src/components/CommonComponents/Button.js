import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  return (
    <div onClick={props.handleClickEvent} className={props.styleClasses} name={props.name}>
      {props.content}
    </div>
  );
};

Button.propTypes = {
  handleClickEvent: PropTypes.func.isRequired,
  content: PropTypes.string.isRequired,
  styleClasses: PropTypes.string.isRequired,
  name: PropTypes.string
};

export default Button;
