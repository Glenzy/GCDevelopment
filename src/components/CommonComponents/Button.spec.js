import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('the button component', ()=>{
  beforeAll();
  it('should display correctly', ()=>{
    const handleClickEvent = jest.fn();
    const   content = 'Button test';
    const styleClasses = 'dummyClassA dummyClassB';
    const name = 'Button';
      const tree = renderer.create(
        <Button
          handleClickEvent={handleClickEvent}
          content={content}
          styleClasses={styleClasses}
          name={name}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
  });
});
