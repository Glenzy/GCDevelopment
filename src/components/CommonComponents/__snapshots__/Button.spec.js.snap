// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should display correctly 1`] = `
<div
  className="dummyClassA dummyClassB"
  name="Button"
  onClick={[MockFunction]}
>
  Button test
</div>
`;
